mvg/mvg.py

```diff
@@ -517,6 +518,59 @@ def create_tabular_measurement(
             json=body,
         )
 
+    def list_timestamps(
+        self,
+        sid: str,
+        offset: int = None,
+        limit: int = None,
+        order: SortOrder = SortOrder.ASC.value,
+    ) -> Dict:
+        """
+        Retrieves timestamps for a source.
+        If offset and limit params are not specified, all timestamps will be returned.
+
+        Parameters
+        ----------
+        sid : str
+            source ID.
+        offset: int
+            index of the first timestamp in the database [optional].
+        limit: int
+            maximum number of timestamps to be returned [optional].
+        order : SortOrder
+            Sort order, either "asc" or "desc". Defaults to "asc".
+
+        Returns
+        -------
+        A dictionary containing the following keys:
+
+            - "offset": int, representing starting point of returned items.
+            - "limit: int, representing max items to return."
+            - "items": list of int, representing timestamps.
+            - "total": int, representing total number of items.
+        """
+        logger.info("endpoint %s", self.endpoint)
+        logger.info("retrieving timestamps from source id=%s", sid)
+        url = f"/sources/{sid}/timestamps"
+        params = {}
+        if limit is not None:
+            params["limit"] = limit
+        if offset is not None:
+            params["offset"] = offset
+        if order is not None:
+            params["order"] = order
+        paginated_items = get_paginated_items(self._request, url, params)
+
+        logger.info("%s timestamps in database", paginated_items["total"])
+        logger.info("Returned %s timestamps", len(paginated_items["items"]))
+
+        return {
```

Comment:
why do we return pagination info for this method? Does user of MVG need to worry about this? If it is needed we need to change `list_measurements` as well otherwise change this to return only the items.

Reply:
It is needed to be able to implement the pagination on the client side. I agree in also adding it to measurements, but I would like to add the sorting as well.

I suggest doing this as a next step. Before that I would like to discuss adding the query mechanism (usually implemented as a `q` parameter in a REST endpoint). This would not be a breaking change.